lib/verdict/assignment.rb

```diff
@@ -27,25 +23,29 @@ def temporary?
   end
 
   def returning
-    self.class.new(@experiment, @subject_identifier, @group, @created_at)
+    self.class.new(@experiment, @subject, @group, @created_at)
   end
 
   def returning?
     @first.nil?
   end
 
+  def subject_identifier
+    experiment.subject_identifier(subject)
+  end
+
   def handle
     qualified? ? group.handle : nil
   end
 
   def to_sym
     qualified? ? group.to_sym : nil
-  end  
+  end
 
   def as_json(options = {})
     {
       experiment: experiment.handle,
-      subject:    subject_identifier,
+      subject:    experiment.subject_identifier(subject),
```

Comment:
you can use the `subject_identifier` helper here, so this line can stay as it was before.